pkg/ccl/streamingccl/streamclient/span_config_stream_client.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Licensed as a CockroachDB Enterprise file under the Cockroach Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     https://github.com/cockroachdb/cockroach/blob/master/licenses/CCL.txt
+
+package streamclient
+
+import (
+	"context"
+	"net/url"
+
+	"github.com/cockroachdb/cockroach/pkg/ccl/streamingccl"
+	"github.com/cockroachdb/cockroach/pkg/repstream/streampb"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/util/hlc"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/tracing"
+	"github.com/cockroachdb/errors"
+	"github.com/jackc/pgx/v4"
+)
+
+type spanConfigStreamClient struct {
+	pgxConfig    *pgx.ConnConfig
+	srcConn      *pgx.Conn // pgx connection to the source cluster
+	subscription spanConfigStreamSubscription
+}
+
+var _ Client = &spanConfigStreamClient{}
+
+func NewSpanConfigStreamClient(remote *url.URL) (Client, error) {
+	config, err := setupPGXConfig(remote)
+	if err != nil {
+		return nil, err
+	}
+	return &spanConfigStreamClient{
+		pgxConfig: config,
+	}, nil
+}
+
+func (m *spanConfigStreamClient) Create(
```

Comment:
@stevendanna If you like this approach of creating a new client implementation, I'll add some boilerplate testing here.

Reply:
Can you write a few words about what you see as the advantage of this? As opposed to allowing the user to call SetupSpanConfigsStream on the partition_stream_client?

Reply:
just added a paragraph to the commit message

Reply:
definitely open to discussing this though!